addon/components/uni-button.js

```diff
@@ -27,6 +27,10 @@ export default Component.extend({
 
     let result = this.get('action')();
     if (result && result.then) {
+      if (this.get('isDestroyed') || this.get('isDestroying')) {
```

Comment:
you sure this solves your problem?

Because it might happen that the component is only destroyed on the promise resolve.